### PR TITLE
Add `delete` method to Connection

### PR DESCRIPTION
**Pull request by mhssmnn**
Adds the `delete` method which allows disconnecting a connection.

### CODE REVIEW

File: lib/xeroizer/connection.rb [hunk 1]
```diff
@@ -1,31 +1,49 @@
...
+
+    def delete
+      parent.delete("https://api.xero.com/connections/#{id}")
+    end
+
+    def to_h
+      attributes.transform_keys(&:underscore)
     end
     
     def to_h
```
Comment (nnc) [inline lib/xeroizer/connection.rb:49]: @mhssmnn this old `#to_h` should probably be removed now. Thanks for the PR btw!

Comment (mhssmnn) [reply lib/xeroizer/connection.rb]: @nnc thanks, have done

**Review by CloCkWeRX — APPROVED** [review]

**Comment by CloCkWeRX** [comment]
The code looks fine, I'm just missing a bit of context; might leave it with @nnc to decide on merging or not.

**Comment by mhssmnn** [comment]
Basically allows disconnecting connections as per https://developer.xero.com/documentation/oauth2/auth-flow#disconnect – useful in testing the reconnection process.

**Review by nnc — APPROVED** [review]
@CloCkWeRX I do think this functionality is needed within Xeroizer, and not just for testing, but also to allow users to disconnect their Xero org(s) from within my app.

I'm not a fan of mixing `client` and `parent` terms for what's basically the same thing. In this context `client` makes more sense to me. 

But in any case I'd be happy to have this functionality in Xeroizer.

**Pull request merged**